### PR TITLE
SDK and workflow for error codes

### PR DESCRIPTION
**Pull request by wtrocki**
## How this works

1. Error codes are generated as 3th job due to additional timing needed for QA to pickup error updates.
2. Error codes are generated only on API change due to fact that we agreed with teams to change openapi when errors are changed. This is simpler rather than constraining team to have all errors in the single file.
3. Once Job is triggered we fetch all APIs 

## Notes

COS api is not on stage yet but I have added it manually to provide early support.
Solution provided is dedicated to this SDK. Other SDKS will use different patterns.

## Verification

Will work with @riccardo-forina to replace existing error codes and use SDK once merged.

### CODE REVIEW

File: packages/kafka-management-sdk/src/errors.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
+
```
Comment (wtrocki) [inline packages/kafka-management-sdk/src/errors.ts:1]: This is the most relevant part of the PR

**Comment by wtrocki** [comment]
I need to add error handling to examples

**Review by riccardo-forina — APPROVED** [review]
LGTM

**Comment by wtrocki** [comment]
Will wait for @craicoverflow review and integrate that with kas-ui

File: .github/workflows/generate_errors.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        node-version: [14.x]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Fetch Errors JSON files
+        run: ./scripts/fetch-errors.sh
```
Comment (craicoverflow) [inline .github/workflows/generate_errors.yaml:23]: Incorrect path:

```suggestion
        run: ./scripts/errors/fetch-errors.sh
```

File: .github/workflows/generate_errors.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      - name: Generate Errors SDK
+        run: ./scripts/generate-errors.sh
```
Comment (craicoverflow) [inline .github/workflows/generate_errors.yaml:25]: Incorrect path:

```suggestion
        run: ./scripts/errors/generate-errors.sh
```

File: scripts/errors/generate-errors.sh [hunk 4]
```diff
@@ -0,0 +1 @@
+node ./generate-errors.js
```
Comment (craicoverflow) [inline scripts/errors/generate-errors.sh:1]: When I run generate-errors.sh from the root of the project I get an error. You might want to add the path prefix with `pwd`.

```shell
$ ./scripts/errors/generate-errors.sh
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '/Users/ephelan/code/src/github.com/redhat-developer/app-services-sdk-js/generate-errors.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

File: packages/registry-management-sdk/src/errors.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      }).catch((err) => {
+        if(APIErrorCodes.ERROR_5 == err.code) {
+          // Handle error
+        }
+      })
+      ```
+  */
+  export enum APIErrorCodes  {
+    /** Unspecified error*/
+  ERROR_1 = "SRS-MGMT-1",
```
Comment (craicoverflow) [inline packages/registry-management-sdk/src/errors.ts:17]: The enum key should be indented IMO.

**Review by craicoverflow — CHANGES_REQUESTED** [review]
Some errors with the path to scripts that I found when manually running them based off what is described in the GitHub job.

You can also see this won't work when using act:

```shell
act -j generateErrors
...
[Generate Errors SDK/generateErrors] ⭐  Run Fetch Errors JSON files
[Generate Errors SDK/generateErrors]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /Users/ephelan/code/src/github.com/redhat-developer/app-services-sdk-js/workflow/2] user= workdir=
| /Users/ephelan/code/src/github.com/redhat-developer/app-services-sdk-js/workflow/2: line 1: ./scripts/fetch-errors.sh: No such file or directory
[Generate Errors SDK/generateErrors]   ❌  Failure - Fetch Errors JSON files
Error: exit with `FAILURE`: 127
```

File: packages/connector-management-sdk/src/errors.ts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+      }).catch((err) => {
+        if(APIErrorCodes.ERROR_5 == err.code) {
+          // Handle error
+        }
+      })
+      ```
+  */
+  export enum APIErrorCodes  {
+    /** Forbidden to perform this action*/
+  ERROR_4 = "KAFKAS-MGMT-4",
```
[the remark this answers is not in the document]
Comment (wtrocki) [reply packages/connector-management-sdk/src/errors.ts:17]: It is possible but would require extra field on the backend. I have that planned for phase 2 of the SDKs

**Review by craicoverflow — APPROVED** [review]

**Comment by wtrocki** [comment]
Merging so I can do proper release. Already tested it with UI

**Pull request merged**